docs/RFCs/0002-crashtracker-user-visable-log.md

```diff
@@ -0,0 +1,169 @@
+Parsers MUST accept unexpected optional fields, either by ignoring them, or by displaying them as additional data.
+The version number SHOULD be incremented for important optional fields, and MUST be incremented when a required field is added or removed.
+
+### Stacktraces
+Different languages and language runtimes have different representations of a stacktrace.
+The representation below attempts to collect as much information as possible.
+In addition, not all information may be available at crash-time on a given machine.
+For example, some libraries may have been shipped with debug symbols stripped, meaning that the only information available about a given frame may be the instruction pointer (`ip`) address, stored as a hex number "0xDEADBEEF".
+This address may be given as an absolute address, or a `NormalizedAddress`, which can be used by backend symbolication.
+We follow the [blazezym](https://github.com/libbpf/blazesym) format for normalized addresses.
+
+A stacktrace consists of 
+  - `format`:
+    An identifier describing the format of the stack trace.
+    Allows for extensibility to support different stack trace formats.
+    The format described below is identfied using the string "CrashTrackerV1"
+  - `frames`:
+    An array of `StackFrame`, described below
+
+#### StackFrame
+
+NOTE: All of the given fields below are optional.
+
+- **Absolute Addresses**
+  The actual in-memory addresses used in the crashing process.
+  Combined with mapping information, such as from `/proc/self/maps`, and the relevant binaries, this can be used to reconstruct relevant symbols.
+  These fields follow the scheme used by the [backtrace crate](https://docs.rs/backtrace/latest/backtrace/struct.Frame.html)
+  - `ip`:
+    The current instruction pointer of this frame.
+    This is normally the next instruction to execute in the frame, but not all implementations list this with 100% accuracy (but it’s generally pretty close).
+  - `sp`:
+    The current stack pointer of this frame.
+  - `symbol_address`:
+    The starting symbol address of the frame of this function.
+    This will attempt to rewind the instruction pointer returned by ip to the start of the function, returning that value.
+    In some cases, however, backends will just return ip from this function.
+  - `module_base_address`:
+    The base address of the module to which the frame belongs
+- **Relative Addresses**
+  Addresses expressed as an offset into a given library or executable.
+  Can be used by backend symbolication to generate debug names etc.
+  These follow the [blazezym](https://github.com/libbpf/blazesym) format for normalized addresses.
+  - `file_offset`: 
+    The relative offset of the symbol, in the base file
+  - `meta`:
+    Metadata to allow the backend symbolizer to identify the file that symbol is in.
+    Currently, this includes the file type: "Apk", "Elf" or "Unknown", as well as the `path` and `build_id` identifying the file.
```

Comment:
`meta` seems to be closely tied to blazesym implementation, not sure that `Apk` support is something that we need (and `Apk` is just like `Elf` without a buildID).
Perhaps it would make sense to use a more generic and easier to parse structure (one whose fields do not change depending on the type).
Instead of:
```
"meta": {
	"Elf" : {
 		"path": "/usr/share/dotnet/shared/Microsoft.NETCore.App/8.0.0/libcoreclr.so",
		"build_id": [...]
	}
}
```
something like:
```
"meta": {
	"path": "...",
	"build_id": "...",
	"type": "elf" / "pdb",
	"age": xxx,
}
```